### ISSUE TITLE
warning: 'int SHA256_Init(SHA256_CTX*)' is deprecated

### ISSUE THREAD

**Issue by milahu** [Windscribe/Desktop-App#78]
```
cd backend/linux/helper
qmake
make

g++ -c -pipe -O2 -std=gnu++11 -Wall -Wextra -D_REENTRANT -fPIC -DBOOST_BIND_GLOBAL_PLACEHOLDERS -DQT_NO_DEBUG -I. -I/build/source/backend/linux/helper/../../../build-libs/boost/include -I/build/source/backend/linux/helper/../../../build-libs/openssl/include -I/nix/store/pk0ha0fv1wcwj5b8gi17lzajzlxbb3ni-qtbase-5.15.7-dev/mkspecs/linux-g++ -o executablesignature_linux.o ../../../common/utils/executable_signature/executablesignature_linux.cpp
../../../common/utils/executable_signature/executablesignature_linux.cpp: In member function 'virtual bool ExecutableSignaturePrivate::verify(const string&)':
../../../common/utils/executable_signature/executablesignature_linux.cpp:73:16: warning: 'int SHA256_Init(SHA256_CTX*)' is deprecated: Since OpenSSL 3.0 [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wdeprecated-declarations-Wdeprecated-declarations8;;]
```

https://www.openssl.org/docs/manmaster/man3/SHA256_Init.html

> All of the functions described on this page except for SHA1(), SHA224(), SHA256(), SHA384() and SHA512() are deprecated. Applications should instead use [EVP_DigestInit_ex(3)](https://www.openssl.org/docs/manmaster/man3/EVP_DigestInit_ex.html), [EVP_DigestUpdate(3)](https://www.openssl.org/docs/manmaster/man3/EVP_DigestUpdate.html) and [EVP_DigestFinal_ex(3)](https://www.openssl.org/docs/manmaster/man3/EVP_DigestFinal_ex.html), or the quick one-shot function [EVP_Q_digest(3)](https://www.openssl.org/docs/manmaster/man3/EVP_Q_digest.html). SHA1(), SHA224(), SHA256(), SHA384(), and SHA256() can continue to be used.

https://stackoverflow.com/a/40155962

https://stackoverflow.com/a/34289358

**Comment by bernerdad** [comment]
Hi.  We'll be staying on OpenSSL 1.1.1 until:
- Qt switches to using OpenSSL 3 as their default, scheduled for Qt 6.5.
- We have a need to upgrade from Qt 6.3.1 to 6.5+.  We typically only update to a newer version of Qt if a Qt bug significantly impacting our users has been patched.
- OpenSSL 1.1.1 is no longer being actively maintained.

**Comment by bernerdad** [comment]
Thank you for your PR with cmake, qt6, and openssl3!  We ported the app to Qt6 and cmake as of the v2.5 release.  We've fallen behind with keeping this Github repo up-to-date with our internal Gitlab repo due to a mirroring bug between Github and Gitlab.  This will be rectified soon (a month or so I'm hoping) and this Github repo will closely follow our internal repo.

**Comment by milahu** [comment]
thanks for clarification

> We've fallen behind with keeping this Github repo up-to-date with our internal Gitlab repo due to a mirroring bug between Github and Gitlab.

just git push...?

```sh
git checkout master
git pull internal --tags
git push github --tags --force
```

> Thank you for your PR with cmake, qt6, and openssl3

if i would do it again, i would replace cmake with meson (newer, better)

**Comment by bernerdad** [comment]
OpenSSL SHA deprecation warnings resolved in the 2.8 release.
